src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenIfOperator.vb

```diff
@@ -1247,5 +1247,125 @@ Friend Module BIFOpResult0011mod
 }]]>)
         End Sub
 
+        <Fact>
+        Public Sub IfOnUnconstrainedMethodTypeParameter()
+            CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Friend Module Mod1
+    Sub M1(Of T)(arg1 As T, arg2 As T)
+        System.Console.WriteLine(If(arg1, arg2))
+    End Sub
+
+    Sub M2(Of T1, T2 As T1)(arg1 as T1, arg2 As T2)
```

Comment:
>Sub M2(Of T1, T2 As T1)(arg1 as T1, arg2 As T2) [](start = 4, length = 47)

Consider verifying behavior:
- int, int
- object (null/not-null), int

Same for the corresponding C# test. Also, it doesn't look like C# test verifies IL

Reply:
> - object (null/not-null), int

I guess this is not interesting, this one is - object, int? (null/not-null)

---
In reply to: [217561188](https://github.com/dotnet/roslyn/pull/29724#discussion_r217561188) [](ancestors = 217561188)